_posts/Processing Data/2020-03-09-Exactly-Once-Example.md

```diff
@@ -32,81 +32,116 @@ from the latest Flink checkpoint in case of failures.
-```
-
-
-In another window, start the ExactlyOnceWriter with EXACTLY_ONCE mode set to false.
-This is to demonstrate what happens when Pravega EXACTLY_ONCE mode not enabled 
-
-```
-$ bin/exactlyOnceWriter --controller tcp://localhost:9090 --scope examples --stream mystream --exactlyonce false
-```
-
-Snippets of output shown below:
+Besides, you also need to get the latest Flink binary from [Apache download page](https://flink.apache.org/downloads.html). Follow [this](https://ci.apache.org/projects/flink/flink-docs-stable/getting-started/tutorials/local_setup.html) tutorial to start a local Flink cluster. 
 
-```
-......
+##### 2. Build `pravega-samples` Repository
 
-Start checkpointing at position 16
-Complete checkpointing at position 16
-Artificial failure at position 26
-05/02/2018 17:02:02	Source: Custom Source -> Map(1/1) switched to FAILED 
-io.pravega.examples.flink.primer.util.FailingMapper$IntentionalException: artificial failure
+`pravega-samples` repository provides code samples to connect analytics engines Flink with Pravega as a storage substrate for data streams. It has divided into sub-projects(`pravega-client-examples`, `flink-connector-examples` and `hadoop-connector-examples`), each one addressed to demonstrate a specific component. To build `pravega-samples` from source, follow the [instructions](https://github.com/pravega/pravega-samples#pravega-samples-build-instructions) to use the built-in gradle wrapper.  
 
-......
+##### 3. Start the Pravega Exactly Once Example program
 
-Restore from checkpoint at position 16
-Start checkpointing at position 50
-Complete checkpointing at position 50
+There are multiple ways to run the program in Flink environment including submitting from terminal or Flink UI. Follow the latest instruction from the [github README](https://github.com/pravega/pravega-samples/tree/master/flink-connector-examples/doc/exactly-once) to learn more about the IDE setup and running process.
 
-......
+#### B. Running example on Dell EMC Streaming Data Platform
+##### 1. Change the exactly-once example code to satisfy the SDP running requirements
+Since the original flink connector exactly-once example was designed to run on a standalone Pravega and Flink environment, the code used the `createScope` method from the StreamManager Interface in Pravega. However, due to the security reason, Dell EMC Streaming Data Platform does not allow to create a scope from the code. Please **comment out** `createScope` method from your code.   
+There is one occurrence of `createScope` method in this exactly-once example which locates on following Java file:   
+I. ```pravega-samples/flink-connector-examples/src/main/java/io/pravega/example/flink/Utils.java```  
 
+##### 2. Follow [this post]({{site.baseurl}}/getting started/2020/07/14/create-flink-project-on-streaming-data-platform.html) to learn how to create Flink projects and run on Dell EMC Streaming Data Platform
```

Comment:
@keedya Here is the link for the Flink project deployment instruction: https://pravega.github.io/workshop-samples/getting%20started/2020/07/14/create-flink-project-on-streaming-data-platform.html. In the post, it can be accessed through a hyperlink.

Reply:
@youminhan that looks good; what about the charts option using helm?

Reply:
To be consistent with other posts, I did not add the charts option using helm. Since the code sample uses the Pravega-Sample repo, it may not be possible to add the helm chart to the code repo. However, I can mention it in the post but it will be a lot of extra works for user/customer to follow because they need to clone the repo, install the helm chart, create a new file, copy-paste the helm chart from our code hub website, and run the helm chart. If it's necessary, I can open a new story by adding helm charts options to every code sample posts and general Flink project deployment instruction.

Reply:
@youminhan  sounds good